File: Python/백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.py
N, M = map(int, input().split())
checker = [[] for _ in range(N)]
result = []
for i in range(N):
    checker[i] = input()

for i in range(N-7):
    for j in range(M-7):
        w = 0
        b = 0
        for k in range(i,i+8):
            for l in range(j,j+8):
                if (k+l)%2 == 0:
                    if checker[k][l] != 'W':
                        w += 1
                    else:
                        b += 1
                else:
                    if checker[k][l] != 'B':
                        w += 1
                    else:
                        b += 1
        result.append(w)
        result.append(b)

print(min(result))

File: Java/백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine()," ");
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        char[][] checker = new char[N][M];
        for (int i = 0; i < N; i++) {
            String line = br.readLine();
            checker[i] = line.toCharArray();
        }

        int ans = Integer.MAX_VALUE;

        for (int i = 0; i <= N - 8; i++) {
            for (int j = 0; j <= M - 8; j++) {
                ans = Math.min(ans, paint(checker, i, j));
            }
        }

        System.out.println(ans);
    }

    public static int paint(char[][] board, int startX, int startY) {
        int whiteStart = 0; // W로 시작하는 경우
        int blackStart = 0; // B로 시작하는 경우

        for (int i = 0; i < 8; i++) {
            for (int j = 0; j < 8; j++) {
                char current = board[startX + i][startY + j];
                if ((i + j) % 2 == 0) {
                    if (current != 'W') whiteStart++;
                    if (current != 'B') blackStart++;
                } else {
                    if (current != 'B') whiteStart++;
                    if (current != 'W') blackStart++;
                }
            }
        }

        return Math.min(whiteStart, blackStart);
    }
}
